packages/app/src/app/overmind/effects/api/apiFactory.ts

```diff
@@ -93,67 +67,110 @@ export default (() => {
-          headers: createHeaders(_options.provideJwtToken()),
+          headers: createHeaders(config.provideJwtToken()),
         })
         .then(response => handleResponse(response, options))
         .catch(e => handleError(e));
     },
-    post<T>(path, body, options?): Promise<T> {
+    post(path, body, options) {
       return axios
         .post(API_ROOT + path, decamelizeKeys(body), {
-          headers: createHeaders(_options.provideJwtToken()),
+          headers: createHeaders(config.provideJwtToken()),
         })
         .then(response => handleResponse(response, options))
         .catch(e => handleError(e));
     },
-    patch<T>(path, body, options?): Promise<T> {
+    patch(path, body, options) {
       return axios
         .patch(API_ROOT + path, decamelizeKeys(body), {
-          headers: createHeaders(_options.provideJwtToken()),
+          headers: createHeaders(config.provideJwtToken()),
         })
         .then(response => handleResponse(response, options))
         .catch(e => handleError(e));
     },
-    put<T>(path, body, options?): Promise<T> {
+    put(path, body, options) {
       return axios
         .put(API_ROOT + path, decamelizeKeys(body), {
-          headers: createHeaders(_options.provideJwtToken()),
+          headers: createHeaders(config.provideJwtToken()),
         })
         .then(response => handleResponse(response, options))
         .catch(e => handleError(e));
     },
-    delete<T>(path, params?, options?): Promise<T> {
+    delete(path, params, options) {
       return axios
         .delete(API_ROOT + path, {
           params,
-          headers: createHeaders(_options.provideJwtToken()),
+          headers: createHeaders(config.provideJwtToken()),
         })
         .then(response => handleResponse(response, options))
         .catch(e => handleError(e));
     },
-    request(options) {
+    request(requestConfig, options) {
       return axios
         .request(
-          Object.assign(options, {
-            url: API_ROOT + options.url,
-            body: options.body ? camelizeKeys(options.body) : null,
-            headers: createHeaders(_options.provideJwtToken()),
+          Object.assign(requestConfig, {
+            url: API_ROOT + requestConfig.url,
+            data: requestConfig.data ? camelizeKeys(requestConfig.data) : null,
+            headers: createHeaders(config.provideJwtToken()),
           })
         )
         .then(response => handleResponse(response, options))
         .catch(e => handleError(e));
     },
   };
-})();
+
+  return api;
+};
+
+function getMessage(error: AxiosError) {
+  const response = error.response;
+
+  if (!response || response.status >= 500) {
+    logError(error);
+  }
+
+  if (response && response.data) {
+    if (response.data.errors) {
+      const errors = values(response.data.errors)[0];
+      if (Array.isArray(errors)) {
+        if (errors[0]) {
+          error.message = errors[0]; // eslint-disable-line no-param-reassign
```

Comment:
@christianalfoni Why don't you just return the new value instead of reassigning `error.message` and returning it at the end of the function?  🤔